daemon/api_download.go

```diff
@@ -63,20 +64,39 @@ func postSnapDownload(c *Command, r *http.Request, user *auth.UserState) Respons
 		return BadRequest("download operation requires action")
 	}
 
+	var opts *snapRevisionOptions
+	if len(action.Options) == 1 {
+		opts = &action.Options[0]
+	}
+	if len(action.Options) > 1 {
```

Comment:
Nitpick: I'd replace the order of checks, first bail out on len(action.Options) > 1 and then len(action.Options) == 1